### PR TITLE
Remove QT code

### PR DESCRIPTION
**Pull request by daira**
This removes the QT wallet code, and updates license documentation accordingly.

### CODE REVIEW

**Comment by daira** [comment]
@zkbot try

**Comment by daira** [comment]
@zkbot try

**Comment by bitcartel** [comment]
Bumping to 1.0.7

**Comment by bitcartel** [comment]
Rebased the PR, after fixing rebase conflicts and running full test suite.

**Comment by bitcartel** [comment]
Bumping to 1.0.7 as out of time for other engineers to review this PR.

**Comment by arcalinea** [comment]
We're merging the changes pulled in from upstream first, then rebasing and merging this one after.

**Comment by daira** [comment]
I'm going to wait until #2099, #2100 and #2101 have been merged before rebasing this.

**Comment by arcalinea** [comment]
#2100 has not merged yet, so I'm reluctantly bumping this to 1.0.8. 

We should prioritize this and merge it right after #2100, before any more changes are pulled in from upstream.

**Comment by daira** [comment]
Are you sure this can't sneak in? #2100 has merged now, and I just rebased this.

**Comment by daira** [comment]
@zkbot try

**Comment by daira** [comment]
Oh, the 1.0.7 release PR has already gone in. So yes, this will have to be in 1.0.8.

File: contrib/debian/copyright [hunk 1]
```diff
@@ -4,67 +4,17 @@ Upstream-Contact: Zcash Company <team@z.cash>
...
 License: Expat
 Comment: The Bitcoin Core developers encompasses the current developers listed on bitcoin.org,
  as well as the numerous contributors to the project.
  The Zcash developers are listed at https://z.cash/team.html.
 
-Files: src/qt/res/icons/clock*.png, src/qt/res/icons/tx*.png,
-       src/qt/res/src/*.svg
-Copyright: Wladimir van der Laan
+Files: src/json/*
+Copyright: 2007-2009, John W. Wilkinson
```
Comment (str4d) [inline contrib/debian/copyright]: This was removed in #1990.

File: contrib/devtools/symbol-check.py [hunk 2]
```diff
@@ -53,21 +53,17 @@
...
+'libgomp.so.1', # OpenMP support library
 'ld-linux-x86-64.so.2', # 64-bit dynamic linker
 'ld-linux.so.2', # 32-bit dynamic linker
-# bitcoin-qt only
-'libX11-xcb.so.1', # part of X11
-'libX11.so.6', # part of X11
-'libxcb.so.1', # part of X11
-'libfontconfig.so.1', # font support
-'libfreetype.so.6', # font parsing
 'libdl.so.2' # programming interface to dynamic linker
```
Comment (str4d) [inline contrib/devtools/symbol-check.py]: Is this not `bitcoin-qt` only?

Comment (daira) [inline contrib/devtools/symbol-check.py]: zcashd does appear to use it. Perhaps via a dependency we added? I don't know whether it should.

**Comment by daira** [comment]
Good point about files under depends/.

File: contrib/debian/copyright (hunk 1 above)
Comment (daira) [inline contrib/debian/copyright]: So it was. Will fix.

**Comment by daira** [comment]
@str4d please re-review.

@zkbot try

**Comment by daira** [comment]
(The only things that have changed are https://github.com/zcash/zcash/pull/1636/commits/71d0bbc6665eca298bb2d93db4f58827b4e7f19a#diff-1d348d46a200709d8ac7193d77df3b51 and the last commit.)

**Comment by bitcartel** [comment]
@daira I'm reviewing

**Comment by daira** [comment]
utACK 07cf426404b491e67d4657320d2272914189d447

File: src/init.cpp [hunk 3]
```diff
@@ -487,18 +481,7 @@ std::string HelpMessage(HelpMessageMode mode)
...
-    if (mode == HMM_BITCOIN_QT)
-    {
-        strUsage += HelpMessageGroup(_("UI Options:"));
-        if (showDebug) {
-            strUsage += HelpMessageOpt("-allowselfsignedrootcertificates", "Allow self signed root certificates (default: 0)");
-        }
-        strUsage += HelpMessageOpt("-choosedatadir", _("Choose data directory on startup (default: 0)"));
-        strUsage += HelpMessageOpt("-lang=<lang>", _("Set language, for example \"de_DE\" (default: system locale)"));
-        strUsage += HelpMessageOpt("-min", _("Start minimized"));
-        strUsage += HelpMessageOpt("-rootcertificates=<file>", _("Set SSL root certificates for payment request (default: -system-)"));
```
Comment (str4d) [inline src/init.cpp]: This also needs to be removed from `contrib/bitcoind.bash-completion`.

Comment (daira) [inline src/init.cpp]: Will fix.

**Comment by daira** [comment]
Need one more ACK.

**Comment by bitcartel** [comment]
@daira Pushed commit.  Some more stragglers.

**Comment by bitcartel** [comment]
@daira Pushed a commit to remove gui flag/options which should now be redundant.  Shouldn't impact dependencies since they have their own build settings - unless there are some global overrides.  Needs review.  Builds fine.

**Comment by daira** [comment]
utACK abecfc422cfe4b50451c9a6d71e5a7729950fd96

**Comment by daira** [comment]
@str4d Please also review @bitcartel's commits.

**Comment by str4d** [comment]
That's 2 reviews on everything. If we missed removing anything else, it can go in a new PR.

@zkbot r+

**Comment by daira** [comment]
Woohoo!

**Comment by arcalinea** [comment]
:smile:

**Comment by daira** [comment]
http://68.media.tumblr.com/ae57a748eaffe6ec3a4a2f379974e5f8/tumblr_inline_nlhketaeh51smedza.gif

**Pull request merged**